proposals/0345-android-12-issues.md

```diff
@@ -165,19 +268,67 @@ private static SdlDeviceListener getSdlDeviceListener(Context context, Bluetooth
+We also need to consider interactions with applications supporting older versions of the `SdlRouterService`. If an application using SDL Android 4.11 or older receives an `ACL_CONNECTED` broadcast in the `SdlBroadcastReceiver`, it may try to start up the `SdlRouterService` of an application that is targeting Android 12. If the started `SdlRouterService` does not have Bluetooth permission, it will not be able to use the `bluetoothTransport`. To solve this issue we can update the `initCheck` within the `SdlRouterService` to check the app's Bluetooth permission. If the permission is not granted, we can fail the `initCheck` and try to deploy the next `SdlRouterService`.
 
-If a phone connects to the head unit over USB we can still start the `SdlRouterService`, but if the designated app does not have Bluetooth permissions, the `bluetoothTransport` will still not be able to start. This could cause confusion for the user if they expect the `SdlRouterService` to connect over USB and Bluetooth but the `SdlRouterService` will only connect over USB. The suggested solution is to present a notification to the user reminding them to enable Bluetooth permissions. By clicking on the notification the user will be directed to the app's settings page to grant those permissions. Meanwhile the `SdlRouterService` will wait to initialize the `bluetoothTransport` and will continuously check the permission status. Once the permissions are granted, the `bluetoothTransport` will be started.
+If a phone connects to the head unit over USB we can still start the `SdlRouterService`, but if the designated app does not have Bluetooth permission, the `bluetoothTransport` will still not be able to start. This could cause confusion for the user if they expect the `SdlRouterService` to connect over USB and Bluetooth but the `SdlRouterService` will only connect over USB. The suggested solution is to present a notification to the user reminding them to enable Bluetooth permission. By clicking on the notification the user will be directed to the app's settings page to grant this permission. Meanwhile the `SdlRouterService` will wait to initialize the `bluetoothTransport` and will continuously check the permission status. Once the permissions are granted, the `bluetoothTransport` will be started.
+
+Additionally the `MultiplexBluetoothTransport.java` class is using `BluetoothAdapter.cancelDiscovery()`, This requires an additional Runtime Permission of `BLUETOOTH_SCAN`. There is only one use case where this will be called and that is when the the `SdlRouterService` receives a `ROUTER_REQUEST_BT_CLIENT_CONNECT` message. This is never used in a production setting and as such we can check for the `BLUETOOTH_SCAN` permission when it is called but we do not need to require developers to have this permission.
```

Comment:
```suggestion
Additionally the `MultiplexBluetoothTransport.java` class is using `BluetoothAdapter.cancelDiscovery()`. This requires an additional Runtime Permission of `BLUETOOTH_SCAN`. There is only one use case where this will be called and that is when the the `SdlRouterService` receives a `ROUTER_REQUEST_BT_CLIENT_CONNECT` message. This is never used in a production setting and as such we can check for the `BLUETOOTH_SCAN` permission when it is called but we do not need to require developers to have this permission.
```